### PR TITLE
🔥 Remove `FetchableObject`

### PR DESCRIPTION
**Pull request by danielvdao**
### Context

A PR addressing this issue: https://github.com/ruby-grape/grape-entity/issues/368

The thing I am confused about is why we check if `object.respond_to?(:fetch, true)`. Implementing `#fetch` is tricky and I think that most of the time (if not always) we should rely on `Hash` to implement `#fetch` as that's in Ruby stdlib. Otherwise just default to `#public_send`.

Happy to close if it doesn't make sense either - but wanted to give it a shot and it looks like all tests still pass.

cc @LeFnord

### CODE REVIEW

**Comment by danielvdao** [comment]
Thanks, I'll take care of it 🔥

**Comment by danielvdao** [comment]
@LeFnord done! Let me know if this is what you were looking for.

**Comment by danielvdao** [comment]
Hey @LeFnord some of the Rubocop violations are going off, but I don't think they're related to my PR. What do you think? 🤔

**Comment by LeFnord** [comment]
mmh, will have an eye on

File: spec/grape_entity/entity_spec.rb [hunk 1]
```diff
@@ -29,9 +29,11 @@
         end
 
         it 'makes sure that :format_with as a proc cannot be used with a block' do
-          # rubocop:disable Style/BlockDelimiters
-          expect { subject.expose :name, format_with: proc {} do p 'hi' end }.to raise_error ArgumentError
-          # rubocop:enable Style/BlockDelimiters
+          expect do
```
Comment (danielvdao) [inline spec/grape_entity/entity_spec.rb:32]: I cleaned this one up, but I wasn't sure about the other development dependency ones.

**Comment by LeFnord** [comment]
I'm on it

**Comment by LeFnord** [comment]
thanks again

**Pull request merged**